internal/rbd/rbd_util.go

```diff
@@ -700,14 +683,17 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 func trashRemoveImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) error {
 	// attempt to use Ceph manager based deletion support if available
 
-	args := []string{
-		"trash", "remove",
-		pOpts.getTrashPath(),
-		"--id", cr.ID,
-		"--keyfile=" + cr.KeyFile,
-		"-m", pOpts.Monitors,
+	ra, err := pOpts.conn.GetRBDAdmin()
+	if err != nil {
+		return err
 	}
-	rbdCephMgrSupported, err := addRbdManagerTask(ctx, pOpts, args)
+
+	log.DebugLog(ctx, "rbd: adding task to remove image %s with id %s from trash", pOpts, pOpts.ImageID)
+
+	ta := ra.Task()
+	_, err = ta.AddTrashRemove(admin.NewImageSpec(pOpts.Pool, pOpts.RadosNamespace, pOpts.ImageID))
```

Comment:
Please introduce a helper to get the `ImageSpec` from `rbdImage`. There should be only a single location where `admin.NewImageSpec()` is called.

```go
func (ri *rbdImage) getImageSpec() *admin.NewImageSpec {
    return admin.NewImageSpec(ri.Pool, ri.RadosNamespace, ri.ImageID)
}
```

Reply:
> Please introduce a helper to get the `ImageSpec` from `rbdImage`. There should be only a single location where `admin.NewImageSpec()` is called.
> 
> ```go
> func (ri *rbdImage) getImageSpec() *admin.NewImageSpec {
>     return admin.NewImageSpec(ri.Pool, ri.RadosNamespace, ri.ImageID)
> }
> ```

Actually task required imageID and flatten requires imageName, at go-ceph level it was decided to have single `imageSpec` but for ceph it is `imageIDSpec` and `imageSpec`. 

We would still need two methods, one for id and one for name. 
I prefer to keep it the same as it is now.

Reply:
ah

> We would still need two methods, one for id and one for name.

That is a little sad. The current approach is hard to read, and (as you see with my comment). What is the reason for passing different parameters to `admin.NewImageSpec()`? In case one of the parameters is not set, can a helper function not detect that?

Reply:
> ah
> 
> > We would still need two methods, one for id and one for name.
> 
> That is a little sad. The current approach is hard to read, and (as you see with my comment). What is the reason for passing different parameters to `admin.NewImageSpec()`?

To avoid two almost similar types at go-ceph level 😂

> In case one of the parameters is not set, can a helper function not detect that?

That would be moving the logic inside a function with a boot parameter.

There's no way to automatically detect it. 
If another level of abstraction with helper function is needed, I'll add it but anyway we would have to decide whether it is imageid(trash ops) or imagename.